src/oss/langchain/middleware.mdx

```diff
@@ -453,20 +456,112 @@ const agent = createAgent({
 ```
 :::
 
+### Model fallback
+
+The `ModelFallbackMiddleware` provides automatic model fallback on errors. This middleware attempts to retry failed model calls with alternative models in sequence. When a model call fails, it tries the next model in the fallback list until either a call succeeds or all models have been exhausted.
+
+**Key features:**
+
+- Automatic retry with fallback models when primary model fails
+- Sequential fallback through multiple models
+- Preserves original request parameters while switching models
+- Configurable with any combination of model strings or instances
+
+**Use Cases:**
```

Comment:
```suggestion
**Use cases:**
```
I'm not convinced this should be a list (can be smooshed into a sentence and maybe better for vertical real estate)